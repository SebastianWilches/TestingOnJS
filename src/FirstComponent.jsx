import PropTypes from 'prop-types'

export const FirstComponent = ({ titulo, parrafo, suma }) => {
    return (
        <>
            <h1>{titulo}</h1>
            <p>{parrafo}</p>
            <p>{suma + 30}</p>
        </>
    )
}

FirstComponent.propTypes = {
    titulo: PropTypes.string.isRequired, //El titulo tiene que ser un String requerido
    parrafo: PropTypes.string, //El parrafo tiene que ser String
    suma: PropTypes.number
}

FirstComponent.defaultProps = {
    titulo: "No hay titulo",
    parrafo: "No hay informaci√≥n de parrafo",
    suma: 1000
}