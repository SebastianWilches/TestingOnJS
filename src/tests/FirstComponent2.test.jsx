import { render, screen } from '@testing-library/react'
import { FirstComponent } from "../FirstComponent"

describe('Pruebas en el componente <FirstComponent/>', () => {
    //Refactorización de las pruebas

    //Variables generales
    const title = 'Titulo de la app';
    const parrafo = 'Lorem del parrafo';

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<FirstComponent titulo={title} parrafo={parrafo} />);
        expect(container).toMatchSnapshot();

    })

    test('El componente deberia contener el titulo "Titulo de la app"', () => {

        render(<FirstComponent titulo={title} parrafo={parrafo} />);
        expect(screen.getByText(title)).toBeTruthy();

    })

    test('El H1 debería mostrar información del titulo', () => {

        render(<FirstComponent titulo={title} parrafo={parrafo} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);

    })

    test('Debe contener el subtitulo enviado por props 3 veces', () => { 
        
        render(<FirstComponent titulo={title} parrafo={parrafo} />);
        expect(screen.getAllByText(parrafo).length).toBe(3);

     })
})