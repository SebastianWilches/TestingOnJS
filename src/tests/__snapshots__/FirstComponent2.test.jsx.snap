// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <FirstComponent/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Titulo de la app
  </h1>
  <p>
    Lorem del parrafo
  </p>
  <p>
    Lorem del parrafo
  </p>
  <p>
    Lorem del parrafo
  </p>
  <p>
    1030
  </p>
</div>
`;
